### PR TITLE
Update rubygem-foreman_scc_manager to 2.2.0 Foreman 3.3

### PR DESCRIPTION
**Pull request by nadjaheitmann**
(cherry picked from commit d2a38afd5f86a545ab53f20378cdfb8d219653f8)

### CODE REVIEW

**Review by ekohl — CHANGES_REQUESTED** [review]
Could you pick all commits?

**Comment by nadjaheitmann** [comment]
> Could you pick all commits?

Done.

File: packages/katello/rubygem-foreman_scc_manager/rubygem-foreman_scc_manager.spec [hunk 1]
```diff
@@ -4,11 +4,11 @@
 
 %global gem_name foreman_scc_manager
 %global plugin_name scc_manager
-%global foreman_min_version 1.18
-%global katello_min_version 3.7.0
+%global foreman_min_version 3.4
```
[the remark this answers is not in the document]
Comment (nadjaheitmann) [reply packages/katello/rubygem-foreman_scc_manager/rubygem-foreman_scc_manager.spec:7]: This should actually be 3.1, not 3.4.

File: packages/katello/rubygem-foreman_scc_manager/rubygem-foreman_scc_manager.spec [hunk 2]
```diff
@@ -25,18 +25,34 @@ Requires: %{?scl_prefix_ruby}ruby
...
 BuildRequires: foreman-assets >= %{foreman_min_version}
 BuildRequires: foreman-plugin >= %{foreman_min_version}
 BuildRequires: %{?scl_prefix}rubygem(foreman-tasks) >= 0.10
 BuildRequires: %{?scl_prefix}rubygem(rails) >= 5.1
 BuildRequires: %{?scl_prefix_ruby}ruby(release)
 BuildRequires: %{?scl_prefix_ruby}ruby
 BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+Requires: ruby
+BuildRequires: ruby
+BuildRequires: rubygems-devel
```
Comment (ekohl) [inline packages/katello/rubygem-foreman_scc_manager/rubygem-foreman_scc_manager.spec:38]: Also redundant from the lines above
```suggestion
```

**Review by ekohl — CHANGES_REQUESTED** [review]

File: packages/katello/rubygem-foreman_scc_manager/rubygem-foreman_scc_manager.spec (hunk 1 above)
[the remark this answers is not in the document]
Comment (nadjaheitmann) [reply packages/katello/rubygem-foreman_scc_manager/rubygem-foreman_scc_manager.spec]: I will keep that in mind

File: packages/katello/rubygem-foreman_scc_manager/rubygem-foreman_scc_manager.spec [hunk 3]
```diff
@@ -35,8 +35,20 @@ BuildRequires: %{?scl_prefix_ruby}rubygems-devel
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
 Provides: foreman-plugin-%{plugin_name} = %{version}
+BuildRequires: %{?scl_prefix}rubygem(katello) >= 3.16
```
Comment (ekohl) [inline packages/katello/rubygem-foreman_scc_manager/rubygem-foreman_scc_manager.spec:38]: You can drop the foreman-tasks and rails deps on lines 30 & 31

**Comment by ekohl** [comment]
Looks like the NPM packages aren't there:
```
DEBUG util.py:598:  Error: No Package found for rh-nodejs10-npm(@babel/core) < 8.0.0
DEBUG util.py:598:  Error: No Package found for rh-nodejs10-npm(@babel/core) >= 7.7.0
DEBUG util.py:598:  Error: No Package found for rh-nodejs10-npm(@theforeman/builder) >= 4.0.0
DEBUG util.py:598:  Error: No Package found for rh-nodejs10-npm(react-intl) < 3.0.0
DEBUG util.py:598:  Error: No Package found for rh-nodejs10-npm(react-intl) >= 2.8.0
```

It really is hard to get this to work, given Foreman 3.3 is EOL so we're not going to make changes to those packages anymore.

If you want this to work, I'd recommend releasing a version that's compatible with the older NPM packages.

**Comment by nadjaheitmann** [comment]
> Looks like the NPM packages aren't there:
> 
> ```
> DEBUG util.py:598:  Error: No Package found for rh-nodejs10-npm(@babel/core) < 8.0.0
> DEBUG util.py:598:  Error: No Package found for rh-nodejs10-npm(@babel/core) >= 7.7.0
> DEBUG util.py:598:  Error: No Package found for rh-nodejs10-npm(@theforeman/builder) >= 4.0.0
> DEBUG util.py:598:  Error: No Package found for rh-nodejs10-npm(react-intl) < 3.0.0
> DEBUG util.py:598:  Error: No Package found for rh-nodejs10-npm(react-intl) >= 2.8.0
> ```
> 
> It really is hard to get this to work, given Foreman 3.3 is EOL so we're not going to make changes to those packages anymore.
> 
> If you want this to work, I'd recommend releasing a version that's compatible with the older NPM packages.

@ekohl  Took me a while to finish this up, sorry about that. Now it seems to work. Is it still possible to merge? Would love to have it in Foreman 3.3 .

**Review by ekohl — APPROVED** [review]
I'd be OK with merging this, but do you need it in a release? For Katello that means we need to sign the RPM and kick of a release.

**Comment by nadjaheitmann** [comment]
> I'd be OK with merging this, but do you need it in a release? For Katello that means we need to sign the RPM and kick of a release.

What exactly do you mean? It would be nice if the package was available in Foreman 3.3/Katello 4.5 repos.

**Comment by ekohl** [comment]
Technically Foreman 3.3 and Katello 4.5 are already end of life, so we don't really publish anything in it anymore. We do have the pipelines in place because it can be useful to deal with upgrade issues. Unlike plugins the Katello releases are GPG signed and that means some manual work. So I can make an exception here, but it'd be preferred to focus on supported releases.

**Comment by nadjaheitmann** [comment]
It is not my intention to put extra work on anybody, so next time feel free to remind me if I open PRs against unsupported versions :see_no_evil: For this, we both put in some effort in getting this straight and the PR ready. If you merge it now into foreman-packaging but it does not appear in the repos, then we have a diversion between foreman-packaging and the available packages in https://yum.theforeman.org/katello/4.5/ or am I misunderstanding something?

**Comment by ekohl** [comment]
Correct. So let's merge & release this now and focus on supported releases next time.

**Comment by ekohl** [comment]
Once https://ci.theforeman.org/job/katello-4.5-rpm-pipeline/17/ is green it should be available.

**Comment by nadjaheitmann** [comment]
Thanks :+1:

**Pull request merged**